### PR TITLE
Add new options that control whether host toolchain is included

### PR DESCRIPTION
**Pull request by xtremekforever**
I'm fully prepared for this to be torn apart, but this is hopefully a good "first PR" for this configuration. @euanh @MaxDesiatov 

This adds a new `--no-include-host-toolchain` flag to the `make-linux-sdk` sub-command, making it possible to generate an SDK for x86_64 or aarch64 with only the target sysroot and Swift libraries, like this:

```bash
# x86_64
swift run swift-sdk-generator make-linux-sdk --swift-version 6.0.3-RELEASE --target x86_64-unknown-linux-gnu --no-include-host-toolchain
# aarch64
swift run swift-sdk-generator make-linux-sdk --swift-version 6.0.3-RELEASE --target aarch64-unknown-linux-gnu --no-include-host-toolchain
```

I tested the generated SDKs on both Ubuntu 22.04 and macOS 15 and they work just fine.

Ubuntu x86_64 -> aarch64:

```bash
> swift build --swift-sdk 6.0.3-RELEASE_ubuntu_jammy_aarch64
Building for debugging...
warning: Could not read SDKSettings.json for SDK at: ~/.swiftpm/swift-sdks/6.0.3-RELEASE_ubuntu_jammy_aarch64.artifactbundle/6.0.3-RELEASE_ubuntu_jammy_aarch64/aarch64-unknown-linux-gnu/ubuntu-jammy.sdk
[485/485] Linking ConsumePublishServices
Build complete! (25.37s)
```

macOS arm64 -> x86_64:

```bash
>  swift build --swift-sdk 6.0.3-RELEASE_ubuntu_jammy_x86_64
Building for debugging...
warning: Could not read SDKSettings.json for SDK at: /Users/xtremek/Library/org.swift.swiftpm/swift-sdks/6.0.3-RELEASE_ubuntu_jammy_x86_64.artifactbundle/6.0.3-RELEASE_ubuntu_jammy_x86_64/x86_64-unknown-linux-gnu/ubuntu-jammy.sdk
<unknown>:0: warning: libc not found for 'x86_64-unknown-linux-gnu'; C stdlib may be unavailable
[1171/1171] Linking ConsumePublishServices
Build complete! (39.43s)
```

Also, is there a need/desire to add some unit testing for this flag?

### CODE REVIEW

**Comment by xtremekforever** [comment]
Update: since both 5.9 and 5.10 require the `supportedTriples` to be included in `info.json`, I've opted to add all the supported host triples when building the SDK with `--no-host-toolchain`:

```json
          "supportedTriples" : [
            "x86_64-unknown-linux-gnu",
            "aarch64-unknown-linux-gnu",
            "x86_64-apple-macos",
            "arm64-apple-macos"
          ],
```

Then, either Linux x86_64/aarch64 or macOS x86_64/arm64 hosts can cross compile with the SDK.

Since the SDK doesn't contain the Swift toolchain, though, the open source Swift toolchain is required to be used on macOS for it to compile. I think this is also the case with the Linux Static SDK.

@euanh @MaxDesiatov please have a look here and see what you think!

File: Sources/GeneratorCLI/GeneratorCLI.swift [hunk 1]
```diff
@@ -107,6 +108,11 @@ extension GeneratorCLI {
     )
     var hostSwiftPackagePath: String? = nil
 
+    @Flag(
+      help: "Don't include host toolchain in generated SDK, making it similar to the Static SDKs for Linux."
```
Comment (MaxDesiatov) [inline Sources/GeneratorCLI/GeneratorCLI.swift:112]: This is also the case for the Swift SDK for WASI, thus technically in its current form this help string is incomplete. Regardless, I think it doesn't provide the necessary information to users upfront, they would additionally have to look up what those Swift SDKs do to get the full context.

I suggest rephrasing the help string to explain the benefits of not including the host toolchain in such Swift SDKs.

I would also prefer this to use [`FlagInversion`](https://swiftpackageindex.com/apple/swift-argument-parser/1.5.0/documentation/argumentparser/flaginversion) instead of adding the `no` prefix manually.

**Review by MaxDesiatov — COMMENTED** [review]
Phrasing nit: we should not call these "SDK", but use "Swift SDK" wording instead, since these terms are separate from each other. In the Swift context, plain "SDK" is an Xcode-specific term, where its Darwin SDKs have a distinct file system layout that Xcode build system can consume. "Swift SDK" is a SwiftPM feature, unrelated to plain "SDK"s for Darwin. These are different features in different products and are not interchangeable with each other.

It's unfortunate though that swift.org website doesn't make this distinction, but that would have to be resolved with swift.org website maintainers.

File: Sources/GeneratorCLI/GeneratorCLI.swift (hunk 1 above)
Comment (xtremekforever) [reply Sources/GeneratorCLI/GeneratorCLI.swift:112]: Okay, fair. What do you think of this instead?

```
--host-toolchain/--no-host-toolchain
                          Whether or not to include the host toolchain in the Swift SDK.
                          If the host toolchain is not included, this makes the Swift SDK compatible with any host, much like the Static SDKs from
                          swift.org. (default: --host-toolchain)
```

Comment (MaxDesiatov) [reply Sources/GeneratorCLI/GeneratorCLI.swift:112]: Overall LGTM, but I'd phrase it as `exact same version of swift.org toolchain installed` to make it clear that Xcode toolchains are not compatible (they don't have necessary linkers and other tools).

File: Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift [hunk 2]
```diff
@@ -21,9 +21,10 @@ public struct LinuxRecipe: SwiftSDKRecipe {
     case remoteTarball
   }
 
-  public enum HostSwiftSource: Sendable {
+  public enum HostSwiftSource: Sendable, Equatable {
     case localPackage(FilePath)
     case remoteTarball
+    case skip
```
Comment (MaxDesiatov) [inline Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift:27]: IMO `skip` is more appropriate for an action case. If the name of the type was `HostSwiftGeneratorAction`, then you'd allow the action to skip this step. But with the current name something like `preinstalled` feels more appropriate?

```suggestion
    case preinstalled
```

Comment (xtremekforever) [reply Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift]: Okay that does make sense! `preinstalled` sounds good to me!

File: Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift [hunk 3]
```diff
@@ -240,24 +250,35 @@ public struct LinuxRecipe: SwiftSDKRecipe {
...
+        try await generator.createSymlink(at: autolinkExtractPath, pointingTo: "swift")
+      }
+    } else {
+      // Add all triples that Swift currently supports
+      hostTriples = [
+        Triple("x86_64-unknown-linux-gnu"),
+        Triple("aarch64-unknown-linux-gnu"),
+        Triple("x86_64-apple-macos"),
+        Triple("arm64-apple-macos"),
+      ]
```
[the remark this answers is not in the document]
Comment (xtremekforever) [reply Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift:279]: This doesn't work when generating the Swift SDK for Swift 5.9 and 5.10. Unfortunately they require the `supportedTriples` to be defined or they error out. To demonstrate I set the `hostTriples` above to `nil` and generated the Swift SDK for 5.9.2. Here is the info file:

```json
{
  "schemaVersion" : "1.0",
  "artifacts" : {
    "5.9.2-RELEASE_ubuntu_jammy_x86_64" : {
      "type" : "swiftSDK",
      "version" : "0.0.1",
      "variants" : [
        {
          "path" : "5.9.2-RELEASE_ubuntu_jammy_x86_64/x86_64-unknown-linux-gnu"
        }
      ]
    }
  }
}
```

And Swift 5.9.2 throws this error:

```
warning: Couldn't parse `info.json` manifest of a destination bundle at ~/swift-sdk-generator/Bundles/5.9.2-RELEASE_ubuntu_jammy_x86_64.artifactbundle: failed parsing ArtifactsArchive info.json at '~/swift-sdk-generator/Bundles/5.9.2-RELEASE_ubuntu_jammy_x86_64.artifactbundle/info.json': keyNotFound(CodingKeys(stringValue: "supportedTriples", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "artifacts", intValue: nil), _JSONKey(stringValue: "5.9.2-RELEASE_ubuntu_jammy_x86_64", intValue: nil), CodingKeys(stringValue: "variants", intValue: nil), _JSONKey(stringValue: "Index 0", intValue: 0)], debugDescription: "No value associated with key CodingKeys(stringValue: \"supportedTriples\", intValue: nil) (\"supportedTriples\").", underlyingError: nil))
error: No valid Swift SDK bundles found at ~/swift-sdk-generator/Bundles.
```

This is why I added all the host triples instead of making it nil.

File: Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift [hunk 4]
```diff
@@ -24,6 +24,7 @@ public actor SwiftSDKGenerator {
   let isIncremental: Bool
   let isVerbose: Bool
   let engineCachePath: SQLite.Location
+  let includeHostToolchain: Bool
```
[the remark this answers is not in the document]
Comment (xtremekforever) [reply Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift:27]: Where would you suggest it to go? As you can see, this flag is being used in several places in the `SwiftSDKGenerator`, and prevents it from adding host toolchain files and tweaks in various places...

File: Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (kateinoigakukun) [reply Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift:279]: Hmm, yes https://github.com/swiftlang/swift-package-manager/pull/7432 is available only after 6.0. If you have 5.9 and 5.10 use case, it's fine to leave as is but could you add FIXME to note that it should be removed once we drop 5.10 support?

[the remark this answers is not in the document]
Comment (MaxDesiatov) [reply Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift:279]: Don't we know the host version when generating a Swift SDK? Can we not use the full list or `nil` conditionally based on the host toolchain version?

[the remark this answers is not in the document]
Comment (xtremekforever) [reply Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift:279]: I can do that as well. I originally had it only set hostTriples to nil when under 6.0 or later. If you like this could be the solution.

**Comment by MaxDesiatov** [comment]
> Also, is there a need/desire to add some unit testing for this flag?

Yes please, that would be much appreciated.

File: Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (kateinoigakukun) [reply Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift:27]: I meant something like this patch:

<details>
```diff
diff --git a/Sources/GeneratorCLI/GeneratorCLI.swift b/Sources/GeneratorCLI/GeneratorCLI.swift
index 6c889de..2d13fd4 100644
--- a/Sources/GeneratorCLI/GeneratorCLI.swift
+++ b/Sources/GeneratorCLI/GeneratorCLI.swift
@@ -36,7 +36,6 @@ struct GeneratorCLI: AsyncParsableCommand {
         artifactID: options.sdkName ?? recipe.defaultArtifactID,
         isIncremental: options.incremental,
         isVerbose: options.verbose,
-        includeHostToolchain: options.includeHostToolchain,
         logger: logger
       )

diff --git a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Download.swift b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Download.swift
index d94cbb7..1910639 100644
--- a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Download.swift
+++ b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Download.swift
@@ -119,10 +119,6 @@ extension SwiftSDKGenerator {
         try await fs.unpack(file: tmpDir.appending(fileName), into: sdkDirPath)
       }
     }
-
-    if self.includeHostToolchain {
-      try createDirectoryIfNeeded(at: pathsConfiguration.toolchainBinDirPath)
-    }
   }

   func downloadFiles(
diff --git a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Entrypoint.swift b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Entrypoint.swift
index 0939a1f..a61e531 100644
--- a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Entrypoint.swift
+++ b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Entrypoint.swift
@@ -41,12 +41,10 @@ public extension SwiftSDKGenerator {
       httpClientType = OfflineHTTPClient.self
       #endif
       try await httpClientType.with { client in
-        if self.includeHostToolchain {
-          if !self.isIncremental {
-            try await self.removeRecursively(at: pathsConfiguration.toolchainDirPath)
-          }
-          try await self.createDirectoryIfNeeded(at: pathsConfiguration.toolchainDirPath)
+        if !self.isIncremental {
+          try await self.removeRecursively(at: pathsConfiguration.toolchainDirPath)
         }
+        try await self.createDirectoryIfNeeded(at: pathsConfiguration.toolchainDirPath)

         try await self.createDirectoryIfNeeded(at: pathsConfiguration.artifactsCachePath)

diff --git a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift
index 4144c68..6a735bf 100644
--- a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift
+++ b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift
@@ -26,23 +26,18 @@ extension SwiftSDKGenerator {

     let toolsetJSONPath = pathsConfiguration.swiftSDKRootPath.appending("toolset.json")

-    var toolset: Toolset
-    if self.includeHostToolchain {
-      var relativeToolchainBinDir = pathsConfiguration.toolchainBinDirPath
-
-      guard
-        relativeToolchainBinDir.removePrefix(pathsConfiguration.swiftSDKRootPath)
-      else {
-        fatalError(
-          "`toolchainBinDirPath` is at an unexpected location that prevents computing a relative path"
-        )
-      }
+    var relativeToolchainBinDir = pathsConfiguration.toolchainBinDirPath

-      toolset = Toolset(rootPath: relativeToolchainBinDir.string)
+++ b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift
@@ -26,23 +26,18 @@ extension SwiftSDKGenerator {

     let toolsetJSONPath = pathsConfiguration.swiftSDKRootPath.appending("toolset.json")

-    var toolset: Toolset
-    if self.includeHostToolchain {
-      var relativeToolchainBinDir = pathsConfiguration.toolchainBinDirPath
-
-      guard
-        relativeToolchainBinDir.removePrefix(pathsConfiguration.swiftSDKRootPath)
-      else {
-        fatalError(
-          "`toolchainBinDirPath` is at an unexpected location that prevents computing a relative path"
-        )
-      }
+    var relativeToolchainBinDir = pathsConfiguration.toolchainBinDirPath

-      toolset = Toolset(rootPath: relativeToolchainBinDir.string)
-    } else {
-      toolset = Toolset(rootPath: nil)
+    guard
+      relativeToolchainBinDir.removePrefix(pathsConfiguration.swiftSDKRootPath)
+    else {
+      fatalError(
+        "`toolchainBinDirPath` is at an unexpected location that prevents computing a relative path"
+      )
     }

+    var toolset = Toolset(rootPath: relativeToolchainBinDir.string)
+
     recipe.applyPlatformOptions(toolset: &toolset, targetTriple: self.targetTriple)
     try writeFile(at: toolsetJSONPath, encoder.encode(toolset))

diff --git a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift
index 7a77fd8..82debf3 100644
--- a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift
+++ b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift
@@ -24,7 +24,6 @@ public actor SwiftSDKGenerator {
   let isIncremental: Bool
   let isVerbose: Bool
   let engineCachePath: SQLite.Location
-  let includeHostToolchain: Bool
   let logger: Logger

   public init(
@@ -33,7 +32,6 @@ public actor SwiftSDKGenerator {
     artifactID: String,
     isIncremental: Bool,
     isVerbose: Bool,
-    includeHostToolchain: Bool,
     logger: Logger
   ) async throws {
     logGenerationStep("Looking up configuration values...")
@@ -58,7 +56,6 @@ public actor SwiftSDKGenerator {
...skipping...
     artifactID: String,
     isIncremental: Bool,
     isVerbose: Bool,
-    includeHostToolchain: Bool,
     logger: Logger
   ) async throws {
     logGenerationStep("Looking up configuration values...")
@@ -58,7 +56,6 @@ public actor SwiftSDKGenerator {
     self.isVerbose = isVerbose

     self.engineCachePath = .path(self.pathsConfiguration.artifactsCachePath.appending("cache.db"))
-    self.includeHostToolchain = includeHostToolchain
     self.logger = logger
   }

diff --git a/Sources/SwiftSDKGenerator/Serialization/Toolset.swift b/Sources/SwiftSDKGenerator/Serialization/Toolset.swift
index 0ba5b07..ad70975 100644
--- a/Sources/SwiftSDKGenerator/Serialization/Toolset.swift
+++ b/Sources/SwiftSDKGenerator/Serialization/Toolset.swift
@@ -26,7 +26,7 @@ public struct Toolset: Encodable {

   /// Root path of the toolset, if present. When filling in ``Toolset.ToolProperties/path``, if a raw path string in
   /// ``DecodedToolset`` is inferred to be relative, it's resolved as absolute path relatively to `rootPath`.
-  let rootPath: String?
+  var rootPath: String?

   // MARK: Tools currently known and used by SwiftPM.

diff --git a/Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift b/Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift
index c1f75e4..f4bfaff 100644
--- a/Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift
+++ b/Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift
@@ -109,6 +109,9 @@ public struct LinuxRecipe: SwiftSDKRecipe {
   }

   public func applyPlatformOptions(toolset: inout Toolset, targetTriple: Triple) {
+    if self.hostSwiftSource == .preinstalled {
+      toolset.rootPath = nil
+    }
     var swiftCompilerOptions = ["-Xlinker", "-R/usr/lib/swift/linux/"]

     // Swift 5.9 does not handle the `-use-ld` option properly:
diff --git a/Tests/SwiftSDKGeneratorTests/ArchitectureMappingTests.swift b/Tests/SwiftSDKGeneratorTests/ArchitectureMappingTests.swift
index 22c1dd3..a06551a 100644
--- a/Tests/SwiftSDKGeneratorTests/ArchitectureMappingTests.swift
+++ b/Tests/SwiftSDKGeneratorTests/ArchitectureMappingTests.swift
@@ -67,7 +67,6 @@ final class ArchitectureMappingTests: XCTestCase {
       // combinations of build and runtime architecture
       isIncremental: false,
       isVerbose: false,
-      includeHostToolchain: true,
       logger: Logger(label: "org.swift.swift-sdk-generator")
     )

swift-sdk-generator (#164-no-host-toolchain-option +$)$ git diff --cached > t.diff
swift-sdk-generator (#164-no-host-toolchain-option +$%)$ cat t.diff
diff --git a/Sources/GeneratorCLI/GeneratorCLI.swift b/Sources/GeneratorCLI/GeneratorCLI.swift
index 6c889de..2d13fd4 100644
--- a/Sources/GeneratorCLI/GeneratorCLI.swift
+++ b/Sources/GeneratorCLI/GeneratorCLI.swift
@@ -36,7 +36,6 @@ struct GeneratorCLI: AsyncParsableCommand {
         artifactID: options.sdkName ?? recipe.defaultArtifactID,
         isIncremental: options.incremental,
         isVerbose: options.verbose,
-        includeHostToolchain: options.includeHostToolchain,
         logger: logger
       )

diff --git a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Download.swift b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Download.swift
index d94cbb7..1910639 100644
--- a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Download.swift
+++ b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Download.swift
@@ -119,10 +119,6 @@ extension SwiftSDKGenerator {
         try await fs.unpack(file: tmpDir.appending(fileName), into: sdkDirPath)
       }
     }
-
-    if self.includeHostToolchain {
-      try createDirectoryIfNeeded(at: pathsConfiguration.toolchainBinDirPath)
-    }
   }

   func downloadFiles(
diff --git a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Entrypoint.swift b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Entrypoint.swift
index 0939a1f..a61e531 100644
--- a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Entrypoint.swift
+++ b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Entrypoint.swift
@@ -41,12 +41,10 @@ public extension SwiftSDKGenerator {
       httpClientType = OfflineHTTPClient.self
       #endif
       try await httpClientType.with { client in
-        if self.includeHostToolchain {
-          if !self.isIncremental {
-            try await self.removeRecursively(at: pathsConfiguration.toolchainDirPath)
-          }
-          try await self.createDirectoryIfNeeded(at: pathsConfiguration.toolchainDirPath)
+        if !self.isIncremental {
+          try await self.removeRecursively(at: pathsConfiguration.toolchainDirPath)
         }
+        try await self.createDirectoryIfNeeded(at: pathsConfiguration.toolchainDirPath)

         try await self.createDirectoryIfNeeded(at: pathsConfiguration.artifactsCachePath)

diff --git a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift
index 4144c68..6a735bf 100644
--- a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift
+++ b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift
@@ -26,23 +26,18 @@ extension SwiftSDKGenerator {

     let toolsetJSONPath = pathsConfiguration.swiftSDKRootPath.appending("toolset.json")

-    var toolset: Toolset
-    if self.includeHostToolchain {
-      var relativeToolchainBinDir = pathsConfiguration.toolchainBinDirPath
-
-      guard
-        relativeToolchainBinDir.removePrefix(pathsConfiguration.swiftSDKRootPath)
-      else {
-        fatalError(
-          "`toolchainBinDirPath` is at an unexpected location that prevents computing a relative path"
-        )
-      }
+    var relativeToolchainBinDir = pathsConfiguration.toolchainBinDirPath

-      toolset = Toolset(rootPath: relativeToolchainBinDir.string)
-    } else {
-      toolset = Toolset(rootPath: nil)
+    guard
+      relativeToolchainBinDir.removePrefix(pathsConfiguration.swiftSDKRootPath)
+    else {
+      fatalError(
+        "`toolchainBinDirPath` is at an unexpected location that prevents computing a relative path"
+      )
     }

+    var toolset = Toolset(rootPath: relativeToolchainBinDir.string)
+
     recipe.applyPlatformOptions(toolset: &toolset, targetTriple: self.targetTriple)
     try writeFile(at: toolsetJSONPath, encoder.encode(toolset))

diff --git a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift
index 7a77fd8..82debf3 100644
--- a/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift
+++ b/Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift
@@ -24,7 +24,6 @@ public actor SwiftSDKGenerator {
   let isIncremental: Bool
   let isVerbose: Bool
   let engineCachePath: SQLite.Location
-  let includeHostToolchain: Bool
   let logger: Logger

   public init(
@@ -33,7 +32,6 @@ public actor SwiftSDKGenerator {
     artifactID: String,
     isIncremental: Bool,
     isVerbose: Bool,
-    includeHostToolchain: Bool,
     logger: Logger
   ) async throws {
     logGenerationStep("Looking up configuration values...")
@@ -58,7 +56,6 @@ public actor SwiftSDKGenerator {
     self.isVerbose = isVerbose

     self.engineCachePath = .path(self.pathsConfiguration.artifactsCachePath.appending("cache.db"))
-    self.includeHostToolchain = includeHostToolchain
     self.logger = logger
   }

diff --git a/Sources/SwiftSDKGenerator/Serialization/Toolset.swift b/Sources/SwiftSDKGenerator/Serialization/Toolset.swift
index 0ba5b07..ad70975 100644
--- a/Sources/SwiftSDKGenerator/Serialization/Toolset.swift
+++ b/Sources/SwiftSDKGenerator/Serialization/Toolset.swift
@@ -26,7 +26,7 @@ public struct Toolset: Encodable {

   /// Root path of the toolset, if present. When filling in ``Toolset.ToolProperties/path``, if a raw path string in
   /// ``DecodedToolset`` is inferred to be relative, it's resolved as absolute path relatively to `rootPath`.
-  let rootPath: String?
+  var rootPath: String?

   // MARK: Tools currently known and used by SwiftPM.

diff --git a/Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift b/Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift
index c1f75e4..f4bfaff 100644
--- a/Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift
+++ b/Sources/SwiftSDKGenerator/SwiftSDKRecipes/LinuxRecipe.swift
@@ -109,6 +109,9 @@ public struct LinuxRecipe: SwiftSDKRecipe {
   }

   public func applyPlatformOptions(toolset: inout Toolset, targetTriple: Triple) {
+    if self.hostSwiftSource == .preinstalled {
+      toolset.rootPath = nil
+    }
     var swiftCompilerOptions = ["-Xlinker", "-R/usr/lib/swift/linux/"]

     // Swift 5.9 does not handle the `-use-ld` option properly:
diff --git a/Tests/SwiftSDKGeneratorTests/ArchitectureMappingTests.swift b/Tests/SwiftSDKGeneratorTests/ArchitectureMappingTests.swift
index 22c1dd3..a06551a 100644
--- a/Tests/SwiftSDKGeneratorTests/ArchitectureMappingTests.swift
+++ b/Tests/SwiftSDKGeneratorTests/ArchitectureMappingTests.swift
@@ -67,7 +67,6 @@ final class ArchitectureMappingTests: XCTestCase {
       // combinations of build and runtime architecture
       isIncremental: false,
       isVerbose: false,
-      includeHostToolchain: true,
       logger: Logger(label: "org.swift.swift-sdk-generator")
     )

```
</details>

[the remark this answers is not in the document]
Comment (xtremekforever) [reply Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator.swift:27]: Okay, I understand now! I will implement your suggestion, then add tests to everything after that. Thank you.

**Review by kateinoigakukun — APPROVED** [review]
Thanks!

**Comment by xtremekforever** [comment]
> > Also, is there a need/desire to add some unit testing for this flag?
> 
> Yes please, that would be much appreciated.

@MaxDesiatov I added tests that check the following pieces of functionality in `LinuxRecipe` that are affected by the `.preinstalled` host Swift enum:
 - Toolset options
 - Downloadable artifacts
 - Host triples (`supportedTriples` in info.json)

**Comment by kateinoigakukun** [comment]
@swift-ci test

File: Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift [hunk 5]
```diff
@@ -27,7 +27,6 @@ extension SwiftSDKGenerator {
     let toolsetJSONPath = pathsConfiguration.swiftSDKRootPath.appending("toolset.json")
 
     var relativeToolchainBinDir = pathsConfiguration.toolchainBinDirPath
-
```
Comment (MaxDesiatov) [inline Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift:30]: nit: unnecessary newline that only adds to the diff as an unrelated change

**Review by MaxDesiatov — APPROVED** [review]
LGTM modulo a super minor formatting nit

File: Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift (hunk 5 above)
Comment (xtremekforever) [reply Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Metadata.swift]: Fixed!

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by xtremekforever** [comment]
All good to merge since everything's in there?

**Comment by MaxDesiatov** [comment]
I wonder if @euanh would have any feedback, since he contributed to the Linux generator a lot recently and also enabled tests for it?

**Comment by euanh** [comment]
I've run the EndToEnd tests.  They didn't find any regressions (takes a while):
```
SWIFT_SDK_GENERATOR_RUN_SLOW_TESTS=1 swift test --disable-swift-testing --filter EndToEndTests
...
Test Suite 'Swift60_UbuntuEndToEndTests' passed at 2025-01-07 10:29:19.900.
         Executed 4 tests, with 0 failures (0 unexpected) in 447.902 (447.902) seconds
Test Suite 'swift-sdk-generatorPackageTests.xctest' passed at 2025-01-07 10:29:19.900.
         Executed 18 tests, with 0 failures (0 unexpected) in 2034.483 (2034.484) seconds
Test Suite 'Selected tests' passed at 2025-01-07 10:29:19.900.
         Executed 18 tests, with 0 failures (0 unexpected) in 2034.483 (2034.485) seconds
```

**Review by euanh — APPROVED** [review]

File: Sources/GeneratorCLI/GeneratorCLI.swift [hunk 6]
```diff
@@ -107,6 +107,16 @@ extension GeneratorCLI {
...
 
+    @Flag(
+      inversion: .prefixedNo,
+      help: """
+      Whether or not to include the host toolchain in the Swift SDK.
+      If the host toolchain is not included, this makes the Swift SDK compatible with any host, \
+      but requires exactly the same version of the swift.org toolchain to be installed for it to work.
+      """
+    )
+    var includeHostToolchain: Bool = true
```
Comment (MaxDesiatov) [inline Sources/GeneratorCLI/GeneratorCLI.swift:118]: Sorry about a request coming this late, just noticed that `--no-include-host-toolchain` reads quite weird to me? Although I'm not a native English speaker, so take this concern with a grain of salt. If you agree that `--no-host-toolchain` reads better, let's use that shorter wording and I'm ready to merge immediately after that.

Comment (euanh) [reply Sources/GeneratorCLI/GeneratorCLI.swift:118]: Yes, `--no-host-toolchain` sounds better.

Comment (xtremekforever) [reply Sources/GeneratorCLI/GeneratorCLI.swift]: Done:

```
  --host-toolchain/--no-host-toolchain
                          Whether or not to include the host toolchain in the Swift SDK.
                          If the host toolchain is not included, this makes the Swift SDK compatible with any host, but requires exactly the same version of the
                          swift.org toolchain to be installed for it to work. (default: --host-toolchain)
```
